convert2rhel/checks.py

```diff
@@ -835,3 +841,81 @@ def check_dbus_is_running():
         "Please start dbus using `systemctl start dbus` or (on CentOS Linux 6), `service messagebus start`"
     )
+
+
+def _is_initramfs_file_valid(filepath):
+    """Internal function to verify if an initramfs file is corrupted.
+
+    This method will rely on using lsinitrd to do the validation. If the
+    lsinitrd returns other value that is not 0, then it means that the file is
+    probably corrupted or may cause problems during the next reboot.
+
+    :param filepath: The path to the initramfs file.
+    :type filepath: str
+    :return: A boolean to determine if the file is corrupted.
+    :rtype: bool
+    """
+    logger.info("Checking if the '%s' file is not corrupted.", filepath)
+    out, return_code = run_subprocess(
+        cmd=["/usr/bin/lsinitrd", filepath],
+        print_output=False,
+    )
+
+    if return_code != 0:
+        logger.info("Couldn't verify initramfs file. It may be corrupted.")
+        logger.debug("Output of lsinitrd: %s", out)
+        return False
+
+    return True
+
+
+def check_kernel_boot_files():
+    """Check if the required kernel files exist and are valid under the boot partition."""
+    # For Oracle/CentOS Linux 8 the `kernel` is just a meta package, instead,
+    # we check for `kernel-core`. This is not true regarding the 7.* releases.
+    kernel_name = "kernel-core" if system_info.version.major >= 8 else "kernel"
+
+    # Either the package is returned or not. The return_code will be 0 in
+    # either case, so we don't care about checking for that here.
+    output, _ = run_subprocess(["rpm", "-q", "--last", kernel_name], print_output=False)
+
+    # We are parsing the latest kernel installed on the system, which at this
+    # point, should be a RHEL kernel. Since we can't get the kernel version
+    # from `uname -r`, as it requires a reboot in order to take place, we are
+    # detecting the latest kernel by using `rpm` and figuring out which was the
+    # latest kernel installed.
+    latest_installed_kernel = output.split("\n")[0].split(" ")[0]
+    latest_installed_kernel = latest_installed_kernel.split("%s-" % kernel_name)[-1]
+    grub2_config_file = grub.get_grub_config_file()
+    initramfs_file = INITRAMFS_FILEPATH % latest_installed_kernel
+    vmlinuz_file = VMLINUZ_FILEPATH % latest_installed_kernel
+
+    logger.debug("Checking if the '%s' file exists", vmlinuz_file)
+    vmlinuz_exists = os.path.exists(vmlinuz_file)
+    logger.debug("Result of vmlinuz presence: %s", vmlinuz_exists)
+
+    # We always set this value to True, as a way of having an initial value.
+    # Always assuming the good path.
+    is_initramfs_valid = True
+    if os.path.exists(initramfs_file):
+        is_initramfs_valid = _is_initramfs_file_valid(initramfs_file)
+    else:
+        # Change it to False as if we don't have an initramfs file, then we
+        # shift this value to also reach the below if-statement and output the
+        # warning message.
+        is_initramfs_valid = False
```

Comment:
If you reverse the initial value, you can get rid of the `else:` clause.

```python
    is_initramfs_valid = False
    if os.path.exists(initramfs_file):
        is_initramfs_valid = _is_initramfs_file_valid(initramfs_file)
```

Probably even better, though, would be to move the `os.path.exists()` check into `_is_initramfs_file_valid()` and then the logic here will be:
```suggestion
    is_initramfs_valid = _is_initramfs_file_valid(initramfs_file)

```